blink/blink-shell.html

```diff
@@ -0,0 +1,276 @@
+
+          return canvas;
+        })(),
+        setStatus: function(text) {
+          if (!Module.setStatus.last) Module.setStatus.last = { time: Date.now(), text: '' };
+          if (text === Module.setStatus.last.text) return;
+          var m = text.match(/([^(]+)\((\d+(\.\d+)?)\/(\d+)\)/);
+          var now = Date.now();
+          if (m && now - Module.setStatus.last.time < 30) return; // if this is a progress update, skip it if too soon
+          Module.setStatus.last.time = now;
+          Module.setStatus.last.text = text;
+          if (m) {
+            text = m[1];
+            progressElement.value = parseInt(m[2])*100;
+            progressElement.max = parseInt(m[4])*100;
+            progressElement.hidden = false;
+            spinnerElement.hidden = false;
+          } else {
+            progressElement.value = null;
+            progressElement.max = null;
+            progressElement.hidden = true;
+            if (!text) spinnerElement.style.display = 'none';
+          }
+          statusElement.innerHTML = text;
+        },
+        totalDependencies: 0,
+        monitorRunDependencies: function(left) {
+          this.totalDependencies = Math.max(this.totalDependencies, left);
+          Module.setStatus(left ? 'Preparing... (' + (this.totalDependencies-left) + '/' + this.totalDependencies + ')' : 'All downloads complete.');
+        }
+      };
+      Module.setStatus('Downloading...');
+      window.onerror = function(event) {
+        // TODO: do not warn on ok events like simulating an infinite loop or exitStatus
+        Module.setStatus('Exception thrown, see JavaScript console');
+        spinnerElement.style.display = 'none';
+        Module.setStatus = function(text) {
+          if (text) console.error('[post-exception status] ' + text);
+        };
+      };
+    </script>
+    {{{ SCRIPT }}}
+    <!-- BLINK SPECIFIC: Load and run the assembly. -->
+    <script type='text/javascript'>
+      function fileLoad(event, filename) {
+        var file = event.target.files[0];
+        var reader = new FileReader();
+        reader.onloadend = function(event) {
+          if(event.target.readyState == FileReader.DONE)
+            FS.writeFile(filename, new Uint8Array(event.target.result), {encoding: 'binary'});
+        };
+        reader.readAsArrayBuffer(file);
+      }
+
+      // Also show stderr in the page
+      Module.printErr = (t) => { Module.print(t); };
+
+      function toCString(s) {
+        var size = Module.lengthBytesUTF8(s) + 1;
+        var ptr = Module._malloc(size);
+        Module.stringToUTF8Array(s, Module.HEAP8, ptr, size);
+        return ptr;
+      }
+
+      function runExecutable() {
+        let args = ['blink', 'executable'].concat(document.getElementById('argsInput').value.split(' '));
+
+        let main = Module.cwrap('main', 'number', ['number', 'number']);
```

Comment:
You can just export `callMain` instead and do `Module.callMain(args);'

Reply:
Doesn't `callMain` twice have some undesired effects?

Reply:
Done.